pkg/ruler/compat.go

```diff
@@ -51,7 +52,9 @@ func (a *appendableAppender) AddFast(l labels.Labels, ref uint64, t int64, v flo
 }
 
 func (a *appendableAppender) Commit() error {
-	if _, err := a.pusher.Push(a.ctx, client.ToWriteRequest(a.samples)); err != nil {
+	ctx, cancel := context.WithTimeout(a.ctx, 15*time.Second)
```

Comment:
Is this timeout path sufficiently central to warrant a command line flag? More specifically I wonder if a sufficiently large cluster would benefit from a configurable timeout on writing back to the ingesters.

Reply:
We could... but it's yet another flag. If people want I am happy to add it though. Also, 15 seconds is a very long time to do a single push, it should maybe default to something lower.

Reply:
Just curious mostly, I'm not terribly concerned either way.